### ISSUE TITLE
Dependencies on older module versions

### ISSUE THREAD

**Issue by cawilliamson** [shoekstra/puppet-owncloud#16]
I'm having a few issues with this module depending on older modules:

Warning: Module 'puppetlabs-apache' (v1.4.1) fails to meet some dependencies:
  'shoekstra-owncloud' (v0.4.0) requires 'puppetlabs-apache' (>= 1.0.1 <= 1.2.0)
Warning: Module 'puppetlabs-apt' (v2.0.1) fails to meet some dependencies:
  'shoekstra-owncloud' (v0.4.0) requires 'puppetlabs-apt' (>= 1.5.0 <= 1.7.0)
Warning: Module 'puppetlabs-mysql' (v3.4.0) fails to meet some dependencies:
  'shoekstra-owncloud' (v0.4.0) requires 'puppetlabs-mysql' (>= 2.3.1 <= 3.1.0)
Warning: Module 'puppetlabs-stdlib' (v4.6.0) fails to meet some dependencies:
  'shoekstra-owncloud' (v0.4.0) requires 'puppetlabs-stdlib' (>= 2.4.0 <= 4.5.0)
Warning: Missing dependency 'stahnma-epel':
  'shoekstra-owncloud' (v0.4.0) requires 'stahnma-epel' (>= 1.0.0 <= 1.0.2)

Sadly some of these module versions do not play well with Debian Jessie and for this reason are a bit problematic for us Debian folks.

Any view to resolving this and updating the module to support newer versions of other Puppet modules it depends on?

Thanks!

**Comment by shoekstra** [comment]
Someone has fixed this in shoekstra/puppet-owncloud#13, I'm just waiting for them to update the tests so that they pass.

Bit swamped with work at the moment but hope to clear the backlog of issues over the weekend and push a new release to the forge.

**Comment by cawilliamson** [comment]
Cool - makes sense!

Another issue which I'm not sure if you're aware of - Debian Jessie (8.0) using the official owncloud repo (managed by puppetforge-apt but it's the same repo your module adds) now uses the following path:

/usr/share/owncloud

I had to hack the params.pp config to point to this directory since it seems to assume it should be installed to /var/www/owncloud. With this change in place the issue is now resolved. Just something to be aware of - though this could be an issue with the way I have things setup. Probably worth firing up a VM and finding out though! :)

**Comment by shoekstra** [comment]
Weird, I did test it locally Vagrant before adding "Supports Debian 8".  Looks like I need to do it again!

I've got a Vagrantfile that has VMs for all supported OSes, I think my most immediate step should be to make this available for people to test against so I can reproduce the issues easily.  That or just get around to writing proper acceptance tests!

Thanks for pointing that out, I've created a separate issue to address the path issue.  Hoping to resolve these over the weekend - paying work comes first I'm afraid, sorry about any inconvenience!
